app/craco.config.js

```diff
@@ -1,11 +1,51 @@
-const { hawtioBackend } = require('@hawtio/backend-middleware')
+const ModuleFederationPlugin = require('webpack/lib/container/ModuleFederationPlugin')
 const MonacoWebpackPlugin = require('monaco-editor-webpack-plugin')
+const { dependencies } = require('./package.json')
+const { hawtioBackend } = require('@hawtio/backend-middleware')
 
 module.exports = {
   webpack: {
+    plugins: {
+      add: [
+        new ModuleFederationPlugin({
+          name: 'app',
+          filename: 'remoteEntry.js',
+          exposes: {
+            './remote': './src/examples/remote',
+          },
+          shared: {
+            ...dependencies,
+            react: {
+              singleton: true,
+              requiredVersion: dependencies['react'],
+            },
+            'react-dom': {
+              singleton: true,
+              requiredVersion: dependencies['react-dom'],
+            },
+            'react-router-dom': {
+              singleton: true,
+              requiredVersion: dependencies['react-router-dom'],
+            },
+            '@hawtio/react': {
+              singleton: true,
+              // Hardcoding needed because it cannot handle yarn 'workspace:*' version
+              requiredVersion: '^0.2.0-dev.4',
```

Comment:
Need to make a note of that somewhere to ensure we keep the versions in sync when we increment after release.

Reply:
Actually, `^0.2.0-dev.4` means all 0.x versions higher than or equal to `0.2.0-dev.4` are compatible, so we won't need to update this so frequently. Adding to that, this is the configuration for the "app" package, which is only for development. So I don't think we need to be so careful about updating it.

On the other hand, updating https://github.com/hawtio/hawtio/blob/3.x/console/craco.config.js would be important as it's part of the actual distribution. But thankfully, the configuration there should look like this, so no need for updates:
```
            '@hawtio/react': {
              singleton: true,
              requiredVersion: dependencies['@hawtio/react'],
            },
```

Reply:
Let me take it as resolved